ts/packages/borsh/src/index.ts

```diff
@@ -120,11 +121,11 @@ class WrappedLayout<T, U> extends LayoutCls<U> {
   }
 }
 
-export function publicKey(property?: string): Layout<PublicKey> {
+export function publicKey(property?: string): Layout<Base58EncodedAddress> {
   return new WrappedLayout(
     blob(32),
-    (b: Buffer) => new PublicKey(b),
-    (key: PublicKey) => key.toBuffer(),
+    (b: Buffer) => base58.deserialize(b)[0] as Base58EncodedAddress,
+    (key: Base58EncodedAddress) => Buffer.from(base58.serialize(key)),
```

Comment:
☠️

Developers need to add a separate package for base58 and manually de/serialize addresses even though pretty much every client is going to use them?